cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -131,9 +123,14 @@ private void downloadFileWithProgressBar(String url, Path target, HttpResponse<I
           fileComplete = true;
         } else {
           bufferedOut.write(data, 0, count);
-          pb.stepBy(count);
+          if (contentLength > 0) {
+            pb.stepBy(count);
+          } else {
+            break;
```

Comment:
this `break` will cause that the download is aborted without writing the file to the end.